### PR TITLE
SNA section; package and book additions; details and fixes

### PR DESCRIPTION
**Pull request by briatte**
__Packages__
- added section on social network analysis
- added haven and jsonlite to data manipulation
- added animation to graphic displays
- added hdtv to HTML widgets
- added a few details on rvest and stringr

__Resources__
- added website and R Graphics Cookbook by Winston Chang
- added R Cookbook by Paul Teetor
- added R Packages book/website by Hadley Wickham
- added two books by Roger D. Peng

__Details__
- added a few URLs
- minor fixes here and there

### CODE REVIEW

**Comment by qinwf** [comment]
Thank you so much! Please read the comments on the code.

File: README.md [hunk 1]
```diff
@@ -142,9 +147,9 @@ A curated list of awesome R frameworks, packages and software. Inspired by [awes
 * [rPython](http://cran.r-project.org/web/packages/rPython/index.html) - Package allowing R to call Python.
 * [runr](https://github.com/yihui/runr) - Run Julia and Bash from R.
 * [RJulia](https://github.com/armgong/RJulia) - R package Call Julia.
-* [RinRuby](https://sites.google.com/a/ddahl.org/rinruby-users/) -  a Ruby library that integrates the R interpreter in Ruby.
+* [RinRuby](https://sites.google.com/a/ddahl.org/rinruby-users/) - a Ruby library that integrates the R interpreter in Ruby.
 * [R.matlab](http://cran.r-project.org/web/packages/R.matlab/index.html) - Read and write of MAT files together with R-to-MATLAB connectivity.
-* [RcppOctave](https://github.com/renozao/RcppOctave) - Seamless Interface to Octave and Matlab.
+* [RcppOctave](https://github.com/renozao/RcppOctave) - Seamless Interface to Octave and Masptlab.
```
Comment (qinwf) [inline README.md]: What is Masptlab?

File: README.md [hunk 2]
```diff
@@ -314,32 +318,44 @@ Angle Regression
...
+* [networkDynamic](https://cran.r-project.org/web/packages/networkDynamic/) - Support for dynamic, (inter)temporal networks.
+* [statnet](http://statnet.org/) - The project behind many R network analysis packages.
+* [ergm](https://cran.r-project.org/web/packages/ergm/index.html), [ergm.count](https://cran.r-project.org/web/packages/ergm.count/), [ergm.userterms](https://cran.r-project.org/web/packages/ergm.userterms/index.html) and [xergm](https://cran.r-project.org/web/packages/xergm/index.html) - Exponential random graph models in R.
+* [latentnet](https://cran.r-project.org/web/packages/latentnet/index.html) - Latent position and cluster models for network objects.
+* [tnet](https://cran.r-project.org/web/packages/tnet/index.html) - Network measures for weighted, two-mode and longitudinal networks ([documentation](http://toreopsahl.com/tnet/)).
+* [rgexf](https://bitbucket.org/gvegayon/rgexf/wiki/Home) - Export network objects from R to [GEXF](http://gexf.net/format/), for manipulation with network software like [Gephi](https://gephi.github.io/) or [Sigma](http://sigmajs.org/).
+
 ## R Development
```
Comment (qinwf) [inline README.md]: What about a blank line?

File: README.md [hunk 3]
```diff
@@ -314,32 +318,44 @@ Angle Regression
 * [ape](http://cran.r-project.org/web/packages/ape/index.html) - Analyses of Phylogenetics and Evolution.
 * [pheatmap](http://cran.r-project.org/web/packages/pheatmap/index.html) - Pretty heatmaps made easy.
 
+## Social Network Analysis
```
Comment (qinwf) [inline README.md]: What about `Network Analysis`? That seems to be more general.

File: README.md [hunk 4]
```diff
@@ -314,32 +318,44 @@ Angle Regression
...
 
 * [renjin](http://www.renjin.org/) - a JVM-based interpreter for R.
 * [pqR](http://www.pqr-project.org/) - a "pretty quick" implementation of R
-* [fastR](https://bitbucket.org/allr/fastr/wiki/Home) - FastR is an implementation of the R Language in Java atop Truffle and Graal. 
+* [fastR](https://bitbucket.org/allr/fastr/wiki/Home) - FastR is an implementation of the R Language in Java atop Truffle and Graal.
 * [riposte](https://github.com/jtalbot/riposte) - a fast interpreter and JIT for R.
 * [TERR](http://spotfire.tibco.com/discover-spotfire/what-does-spotfire-do/predictive-analytics/tibco-enterprise-runtime-for-r-terr) - TIBCO Enterprise Runtime for R.
 * [RRE](http://www.revolutionanalytics.com/) - Revolution R Enterprise.
-* [CXXR](https://www.cs.kent.ac.uk/projects/cxxr/) - Refactorising R into C++.
+* [CXXR](https://www.cs.kent.ac.uk/projects/cxxr/) - Refactorizing R into C++.
```
Comment (qinwf) [inline README.md]: https://www.cs.kent.ac.uk/projects/cxxr/ use `s`

File: README.md [hunk 5]
```diff
@@ -356,18 +372,22 @@ Where to discover new R-esources.
...
+* [R Cookbook](http://www.cookbook-r.com/) - A problem-oriented website that supports the _[R Graphics Cookbook](http://shop.oreilly.com/product/0636920023135.do)_.
 
 ## Books
 
-* [The Art of R Programming](http://shop.oreilly.com/product/9781593273842.do) -  It's a good resource for systematically learning fundamentals such as types of objects, control statements, variable scope, classes and debugging in R.
-* [R in Action](http://www.manning.com/kabacoff2/) - This book aims at all levels of users, with sections for beginning, intermediate and advanced R ranging from "Exploring R data structures" to running regressions and conducting factor analyses.
-* [Use R!](http://www.springer.com/series/6991?detailsPage=titles) - This series of inexpensive and focused books from Springer publish shorter books aimed at practitioners. Books can discuss the use of R in a particular subject area, such as bayesian networks, ggplot2 and Rcpp.
+* _[The Art of R Programming](http://shop.oreilly.com/product/9781593273842.do)_ - It's a good resource for systematically learning fundamentals such as types of objects, control statements, variable scope, classes and debugging in R.
```
Comment (qinwf) [inline README.md]: I recommend not using it `_`.

File: README.md [hunk 6]
```diff
@@ -314,32 +318,44 @@ Angle Regression
...
 * [pheatmap](http://cran.r-project.org/web/packages/pheatmap/index.html) - Pretty heatmaps made easy.
 
+## Social Network Analysis
+*Packages to construct, analyze and visualize network data.*
+
+* [igraph](http://igraph.org/r/) - A collection of network analysis tools.
+* [network](https://cran.r-project.org/web/packages/network/index.html) and [sna](https://cran.r-project.org/web/packages/sna/index.html) - Basic tools to manipulate relational data in R.
+* [networkDynamic](https://cran.r-project.org/web/packages/networkDynamic/) - Support for dynamic, (inter)temporal networks.
+* [statnet](http://statnet.org/) - The project behind many R network analysis packages.
+* [ergm](https://cran.r-project.org/web/packages/ergm/index.html), [ergm.count](https://cran.r-project.org/web/packages/ergm.count/), [ergm.userterms](https://cran.r-project.org/web/packages/ergm.userterms/index.html) and [xergm](https://cran.r-project.org/web/packages/xergm/index.html) - Exponential random graph models in R.
```
Comment (qinwf) [inline README.md]: I suggest that we keep the link to `ergm` and remove the others, and it may look better.

File: README.md [hunk 7]
```diff
@@ -314,32 +318,44 @@ Angle Regression
 * [ape](http://cran.r-project.org/web/packages/ape/index.html) - Analyses of Phylogenetics and Evolution.
 * [pheatmap](http://cran.r-project.org/web/packages/pheatmap/index.html) - Pretty heatmaps made easy.
 
+## Social Network Analysis
+*Packages to construct, analyze and visualize network data.*
+
+* [igraph](http://igraph.org/r/) - A collection of network analysis tools.
+* [network](https://cran.r-project.org/web/packages/network/index.html) and [sna](https://cran.r-project.org/web/packages/sna/index.html) - Basic tools to manipulate relational data in R.
```
Comment (qinwf) [inline README.md]: Move `sna` to another line.

File: README.md [hunk 8]
```diff
@@ -89,39 +94,39 @@ A curated list of awesome R frameworks, packages and software. Inspired by [awes
 * [ggvis](https://github.com/rstudio/ggvis) - Interactive grammar of graphics for R.
 * [Leaflet](http://rstudio.github.io/leaflet/) - One of the most popular JavaScript libraries interactive maps.
 * [MetricsGraphics](http://hrbrmstr.github.io/metricsgraphics/) - Enables easy creation of D3 scatterplots, line charts, and histograms.
-* [networkD3](http://christophergandrud.github.io/networkD3/) - D3 JavaScript Network Graphs from R.
+* [networkD3](http://christophergandrud.github.io/networkD3/) and [ndtv](https://cran.r-project.org/web/packages/ndtv/) - D3 JavaScript Network Graphs from R.
```
Comment (qinwf) [inline README.md]: `ndtv` seems to be a graphic package. What about move it to `Graphic Displays`.

File: README.md [hunk 9]
```diff
@@ -21,6 +21,7 @@ A curated list of awesome R frameworks, packages and software. Inspired by [awes
     - [Bayesian](#bayesian)
     - [Finance](#finance)
     - [Bioinformatics](#bioinformatics)
+    - [Social Network Analysis](social-network-analysis)
```
Comment (qinwf) [inline README.md]: This link do not work for now.

File: README.md [hunk 10]
```diff
@@ -38,11 +39,12 @@ A curated list of awesome R frameworks, packages and software. Inspired by [awes
 * [RStudio](http://www.rstudio.org/) - A powerful and productive user interface for R. Works great on Windows, Mac, and Linux.
 * [Emacs + ESS](http://ess.r-project.org/) - Emacs Speaks Statistics is an add-on package for emacs text editors.
 * [Sublime Text + R-Box](http://github.com/randy3k/R-Box/) - Add-on package for Sublime Text 2/3.
+* [TextMate + r.tmblundle](https://github.com/textmate/r.tmbundle) - Add-on package for [TextMate](http://macromates.com/) 1/2.
```
Comment (qinwf) [inline README.md]: I suggest that we can remove the link to `TextMate` to keep it simple.

File: README.md [hunk 11]
```diff
@@ -314,32 +318,44 @@ Angle Regression
...
+*Packages to construct, analyze and visualize network data.*
+
+* [igraph](http://igraph.org/r/) - A collection of network analysis tools.
+* [network](https://cran.r-project.org/web/packages/network/index.html) and [sna](https://cran.r-project.org/web/packages/sna/index.html) - Basic tools to manipulate relational data in R.
+* [networkDynamic](https://cran.r-project.org/web/packages/networkDynamic/) - Support for dynamic, (inter)temporal networks.
+* [statnet](http://statnet.org/) - The project behind many R network analysis packages.
+* [ergm](https://cran.r-project.org/web/packages/ergm/index.html), [ergm.count](https://cran.r-project.org/web/packages/ergm.count/), [ergm.userterms](https://cran.r-project.org/web/packages/ergm.userterms/index.html) and [xergm](https://cran.r-project.org/web/packages/xergm/index.html) - Exponential random graph models in R.
+* [latentnet](https://cran.r-project.org/web/packages/latentnet/index.html) - Latent position and cluster models for network objects.
+* [tnet](https://cran.r-project.org/web/packages/tnet/index.html) - Network measures for weighted, two-mode and longitudinal networks ([documentation](http://toreopsahl.com/tnet/)).
```
Comment (qinwf) [inline README.md]: we can remove the link to documentation, or move the doc link to the first place.

File: README.md [hunk 12]
```diff
@@ -356,18 +372,22 @@ Where to discover new R-esources.
...
-* [Use R!](http://www.springer.com/series/6991?detailsPage=titles) - This series of inexpensive and focused books from Springer publish shorter books aimed at practitioners. Books can discuss the use of R in a particular subject area, such as bayesian networks, ggplot2 and Rcpp.
+* _[The Art of R Programming](http://shop.oreilly.com/product/9781593273842.do)_ - It's a good resource for systematically learning fundamentals such as types of objects, control statements, variable scope, classes and debugging in R.
+* _[R Cookbook](http://shop.oreilly.com/product/9780596809164.do)_ - A quick and simple introduction to conducting many common statistical tasks with R.
+* _[Exploratory Data Analysis with R](https://leanpub.com/exdata)_, _[R Programming for Data Science](https://leanpub.com/rprogramming)_ and _[Report Writing for Data Science in R](https://leanpub.com/reportwriting)_ - Books written as part of the Johns Hopkins Data Science Specialization.
```
Comment (qinwf) [inline README.md]: This looks better?

* [Johns Hopkins Data Science Specialization Books]() -  Books written as part of the .....
    - [Exploratory Data Analysis with R](https://leanpub.com/exdata) - Exploratory Data Analysis with R.....some more detail about the differences.
    - [R Programming for Data Science](https://leanpub.com/rprogramming) - R Programming for Data Science......

or

[Exploratory Data Analysis with R](https://leanpub.com/exdata), [R Programming for Data Science](https://leanpub.com/rprogramming) and [Report Writing for Data Science in R](https://leanpub.com/reportwriting) - Books written as part of the .

Comment (paternogbc) [inline README.md]: First onw looks nice =)

there is also the new book: [The Art of Data Science](https://leanpub.com/artofdatascience)
which is amazing.
;)

File: README.md (hunk 1 above)
Comment (briatte) [inline README.md]: Oops, typo.

File: README.md (hunk 2 above)
Comment (briatte) [inline README.md]: Where should it go, please?

File: README.md (hunk 3 above)
Comment (briatte) [inline README.md]: Alright.

File: README.md (hunk 4 above)
Comment (briatte) [inline README.md]: Works for me. The document seemed primarily US English, so I harmonized based on that. My spellchecker prefers UK English anyway :)

File: README.md (hunk 5 above)
Comment (briatte) [inline README.md]: OK, took italics out.

File: README.md (hunk 6 above)
Comment (briatte) [inline README.md]: Works for me. The rest of the `ergm` package family is in the "Reverse" section of the [CRAN page](https://cran.r-project.org/web/packages/ergm/).

File: README.md (hunk 7 above)
Comment (briatte) [inline README.md]: Done.

File: README.md (hunk 8 above)
Comment (briatte) [inline README.md]: `ndtv` tends to be more specific than all other packages in that section. I have moved it to the "Network Analysis" section, just after `networkDynamic`, which it complements, but please of course feel free to move it to "Graphic Displays".

File: README.md (hunk 9 above)
Comment (briatte) [inline README.md]: Fixed.

File: README.md (hunk 10 above)
Comment (briatte) [inline README.md]: Alright.

File: README.md (hunk 11 above)
Comment (briatte) [inline README.md]: The CRAN page does not explicitly link to the documentation website: only the internal package docs do. I'll remove the documentation website link.

File: README.md (hunk 12 above)
Comment (briatte) [inline README.md]: Applied first style.

I know about the new book, but it's not focused on R.

**Comment by briatte** [comment]
@qinwf Hi – here's the revised text, based on your comments.

**Comment by qinwf** [comment]
Thanks!

**Comment by briatte** [comment]
I recently compiled an [awesome network analysis list](https://github.com/briatte/awesome-network-analysis) that includes [40+ R packages for network analysis](https://github.com/briatte/awesome-network-analysis#r).

Mentioning it in case you might be interested in linking to it from the [Network Analysis section of the list](https://github.com/qinwf/awesome-R#network-analysis).

**Comment by qinwf** [comment]
Ola! It was on https://github.com/qinwf/awesome-R/commit/ba79c5a63bd599049765c0851f8dbbcf73d3203b .

**Comment by briatte** [comment]
Perfect, thank you :)

**Pull request merged**